### PR TITLE
feat: profile v2

### PR DESCRIPTION
**Pull request by idoshamun**
### Describe what this PR does

Profile page is revamped! This is our new year's surprise: more tech debt! 🎁 

There's no official spec, but you can look at the [figma design](https://www.figma.com/file/Ywxj2MFtLVi2lnRuhgeS80/daily.dev---Mockup?node-id=1842%3A23808&mode=dev)

New features that were added:
* Profile readme - Users can now write markdown that will be featured in their profile
* Cover image - Users can upload a cover image alongside their profile image

The new UI was built as a mobile-first, and I did my best to optimize the experience.

It's deployed to preview3, so you can check that out. There are mobile and tablet+ designs and different edge cases if you visit your profile or someone else's.

Here's a screenshot of my profile on mobile:

It's almost ready, I have some missing bits:
- [x] Tests
- [x] Edge cases support for no squads/no bio/no social links
- [x] When navigating through tabs, the scroll is reset
- [ ] Some more QA and bug hunting

Feel free to give it a high-level review in the meantime, or even just play around

## Events

Nothing new

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [x] MobileL (420px)
- [x] Tablet (656px)
- [x] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

### CODE REVIEW

File: packages/shared/__tests__/fixture/feed.ts [hunk 1]
```diff
@@ -70,7 +70,7 @@ const feed: Connection<Post> = {
     {
       node: {
         id: '0e4005b2d3cf191f8c44c2718a457a1e',
-        title: 'Learn SQL & MongoDb Simultaneously — The Easy Way (Part 1)',
+        title: 'Learn SQL & MongoDb Simultaneously The Easy Way (Part 1)',
```
Comment (idoshamun) [inline packages/shared/__tests__/fixture/feed.ts:73]: The unicode messed my test so I had to remove it

File: packages/shared/src/components/MainLayout.tsx [hunk 2]
```diff
@@ -81,11 +80,6 @@ function MainLayout({
   useAuthVerificationRecovery();
   useNotificationParams();
   useReferralReminder();
-  const handlers = useSwipeableSidebar({
```
Comment (idoshamun) [inline packages/shared/src/components/MainLayout.tsx:84]: Nimrod asked to remove completely the swipes for now

File: packages/shared/src/components/UserBadge.tsx [hunk 3]
```diff
@@ -8,25 +8,34 @@ export type UserBadgeProps = {
...
+  removeMargins?: boolean;
+  disableResponsive?: boolean;
 };
 
 const UserBadge = ({
   className,
   content,
   Icon,
   iconProps,
+  removeMargins,
```
Comment (idoshamun) [inline packages/shared/src/components/UserBadge.tsx:20]: The squad component in the new profile doesn't need the responsiveness

File: packages/shared/src/components/comments/CommentBox.tsx [hunk 4]
```diff
@@ -74,13 +78,18 @@ function CommentBox({
       )}
       data-testid="comment"
     >
+      {linkToComment && (
```
Comment (idoshamun) [inline packages/shared/src/components/comments/CommentBox.tsx:81]: The replies feed need to navigate to the comment on click, just like cards

File: packages/shared/src/components/fields/MarkdownInput/index.tsx [hunk 5]
```diff
@@ -276,6 +276,7 @@ function MarkdownInput(
               )}
               icon={icon}
               onClick={() => uploadRef?.current?.click()}
+              type="button"
```
Comment (idoshamun) [inline packages/shared/src/components/fields/MarkdownInput/index.tsx:279]: This fixed an issue that upon clicking on any icon on the markdown bar it submitted the form

File: packages/shared/src/components/profile/RanksWidget.tsx [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+  >
+    <Rank className="w-8 h-8" rank={rank} colorByRank />
+    <span className="hidden tablet:block ml-1 text-theme-label-tertiary">
+      {rankName}
+    </span>
+    <span className="tablet:ml-auto">{count}</span>
+  </div>
+);
+
+export function RanksWidget({
```
Comment (idoshamun) [inline packages/shared/src/components/profile/RanksWidget.tsx:34]: Copied from the previous implementation

File: packages/shared/src/components/profile/ReadingHeatmapWidget.tsx [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+export interface ReadingHeatmapWidgetProps {
+  fullHistory: boolean;
+  selectedHistoryYear: number;
+  readHistory: UserReadHistory[];
+  before: Date;
+  after: Date;
+  yearOptions: string[];
+}
+
+export function ReadingHeatmapWidget({
```
Comment (idoshamun) [inline packages/shared/src/components/profile/ReadingHeatmapWidget.tsx:57]: Copied from the previous implementation

File: packages/shared/src/components/profile/ReadingTagsWidget.tsx [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+import { MostReadTag } from '../../graphql/users';
+import { ActivityContainer, ActivitySectionHeader } from './ActivitySection';
+import { weeklyGoal } from '../../lib/constants';
+import { ReadingTagProgress } from './ReadingTagProgress';
+
+export interface ReadingTagsWidgetProps {
+  mostReadTags: MostReadTag[];
+}
+
+export function ReadingTagsWidget({
```
Comment (idoshamun) [inline packages/shared/src/components/profile/ReadingTagsWidget.tsx:11]: Copied from the previous implementation

File: packages/shared/src/components/profile/Readme.tsx [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+  const isSameUser = loggedUser?.id === user.id;
+  const readme = user?.readmeHtml;
+
+  // Markdown is supported only in the client due to sanitization
+  const isClient = typeof window !== 'undefined';
+
+  const [editMode, setEditMode] = useState(false);
+
+  const queryKey = generateQueryKey(RequestKey.Readme, user);
+  const { data: remoteReadme, isLoading } = useQuery<{
```
Comment (idoshamun) [inline packages/shared/src/components/profile/Readme.tsx:39]: Fetches the readme as markdown only when needed. Lessons learned from the comments

File: packages/shared/src/graphql/common.ts [hunk 10]
```diff
@@ -124,3 +125,12 @@ export const errorMessage = {
 export interface ReferredUsersData {
   referredUsers: Connection<PublicProfile>;
 }
+
+export interface MutationError {
```
Comment (idoshamun) [inline packages/shared/src/graphql/common.ts:129]: Moved to a shared place

File: packages/shared/src/graphql/posts.ts [hunk 11]
```diff
@@ -1,5 +1,5 @@
 import request, { gql } from 'graphql-request';
-import { Author, Comment, Scout } from './comments';
+import type { Author, Comment, Scout } from './comments';
```
Comment (idoshamun) [inline packages/shared/src/graphql/posts.ts:2]: To prevent circular dependency

File: packages/shared/src/hooks/comments/useDeleteComment.ts [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+  description:
+    'Are you sure you want to delete your comment? This action cannot be undone.',
+  okButton: {
+    title: 'Delete',
+    variant: ButtonVariant.Primary,
+    color: ButtonColor.Cabbage,
+  },
+};
+
+export function useDeleteComment(): UseDeleteCommentRet {
```
Comment (idoshamun) [inline packages/shared/src/hooks/comments/useDeleteComment.ts:34]: Refactored to a standalone hook so we can reuse it in multiple places

File: packages/shared/src/useDynamicHeader.ts [hunk 13]
```diff
@@ -0,0 +1,52 @@
+import { Ref, useEffect, useRef, useState } from 'react';
+
+export type UseDynamicHeaderRet<T extends HTMLElement> = {
+  ref: Ref<T>;
+  progress: number | undefined;
+};
+
+export const useDynamicHeader = <T extends HTMLElement>(
```
Comment (idoshamun) [inline packages/shared/src/useDynamicHeader.ts:8]: Used to implement the sticky header on scroll with smooth transition

File: packages/webapp/pages/[userId]/index.tsx [hunk 14]
```diff
@@ -1,306 +1,104 @@
...
+  const {
+    selectedHistoryYear,
+    setSelectedHistoryYear,
+    before,
+    after,
+    yearOptions,
+    fullHistory,
+  } = useActivityTimeFilter();
+
+  const { data: user } = useQuery(
```
Comment (idoshamun) [inline packages/webapp/pages/[userId]/index.tsx:43]: I introduced this query also here for users who visit their own profile to make sure the readme is up to date and not cached. Otherwise, it creates a weird UX where you update the readme and it still shows the cached version

File: packages/webapp/pages/account/profile.tsx [hunk 15]
```diff
@@ -157,15 +209,6 @@ const AccountProfilePage = (): ReactElement => {
             name="github"
             value={user.github}
           />
-          <AccountTextField
```
Comment (idoshamun) [inline packages/webapp/pages/account/profile.tsx:160]: Requested to remove

File: packages/shared/src/components/MainLayout.tsx (hunk 2 above)
Comment (rebelchris) [reply packages/shared/src/components/MainLayout.tsx:84]: Let's see how long before people start complaining 🤣

File: packages/shared/src/components/ProfilePicture.tsx [hunk 16]
```diff
@@ -103,7 +104,7 @@ function ProfilePictureComponent(
         alt={`${user.username || user.id}'s profile`}
         onError={onError}
         className={classes}
-        loading={isCompanion ? 'eager' : 'lazy'}
+        loading={eager || isCompanion ? 'eager' : 'lazy'}
```
Comment (rebelchris) [inline packages/shared/src/components/ProfilePicture.tsx:107]: We actually prefer/wanted to move all to `nativeLazyLoading` would it not work on this one?

File: packages/shared/src/components/comments/MainComment.tsx [hunk 17]
```diff
@@ -61,15 +65,21 @@ export default function MainComment({
...
     >
       {!editProps && (
         <CommentBox
           {...props}
           comment={comment}
           parentId={comment.id}
-          className={{ container: 'border-b' }}
+          className={{
+            container: 'border-b',
+            ...commentBoxClassName,
```
Comment (rebelchris) [inline packages/shared/src/components/comments/MainComment.tsx:81]: This won't do much right?
className here is a object, we read it as `className.container` etc.
Not sure you need it altogether, if so you need to adjust the passing.

File: packages/shared/src/components/comments/MainComment.tsx [hunk 18]
```diff
@@ -32,6 +35,7 @@ const shouldShowBannerOnComment = (
 
 export default function MainComment({
   className,
+  commentBoxClassName,
```
Comment (rebelchris) [inline packages/shared/src/components/comments/MainComment.tsx:38]: Open for this, but generally we prefer the object method of classNames when more then 1 are needed.

File: packages/shared/src/components/profile/Header.tsx [hunk 19]
```diff
@@ -0,0 +1,81 @@
...
+  const [, onShareOrCopyLink] = useShareOrCopyLink({
+    text: `Check out ${user.name}'s profile on daily.dev`,
+    link: user.permalink,
+    trackObject: () => ({ event_name: 'share profile', target_id: user.id }),
+  });
+
+  return (
+    <header
+      className={classNames('flex items-center px-4 h-12', className)}
+      style={style}
```
Comment (rebelchris) [inline packages/shared/src/components/profile/Header.tsx:35]: We can't do this with tailwind?

File: packages/shared/src/components/profile/Header.tsx [hunk 20]
```diff
@@ -0,0 +1,81 @@
...
+    >
+      {sticky ? (
+        <>
+          <ProfilePicture user={user} nativeLazyLoading size="medium" />
+          <div className="flex flex-col ml-2 typo-footnote">
+            <div className="font-bold">{user.name}</div>
+            <div className="text-theme-label-tertiary">
+              {largeNumberFormat(user.reputation)} Reputation
+            </div>
```
Comment (rebelchris) [inline packages/shared/src/components/profile/Header.tsx:44]: Do you need divs here or would `p` do to be more semantic as they only hold copy.

File: packages/shared/src/components/profile/Header.tsx [hunk 21]
```diff
@@ -0,0 +1,81 @@
...
+        >
+          Edit profile
+        </Button>
+      )}
+      <Button
+        className={classNames('ml-auto', isSameUser && 'tablet:ml-0')}
+        variant={ButtonVariant.Float}
+        size={ButtonSize.Small}
+        icon={<ShareIcon />}
+        onClick={() => onShareOrCopyLink()}
```
Comment (rebelchris) [inline packages/shared/src/components/profile/Header.tsx:66]:
```suggestion
        onClick={onShareOrCopyLink}
```

File: packages/shared/src/components/profile/Header.tsx [hunk 22]
```diff
@@ -0,0 +1,81 @@
...
+      />
+      {isSameUser && (
+        <Button
+          className="tablet:hidden ml-2"
+          variant={ButtonVariant.Float}
+          size={ButtonSize.Small}
+          icon={<SettingsIcon />}
+          tag="a"
+          href={`${process.env.NEXT_PUBLIC_WEBAPP_URL}account/profile`}
+          aria-label="Edit profile"
```
Comment (rebelchris) [inline packages/shared/src/components/profile/Header.tsx:76]: These get added automatic right (from what I recall)

File: packages/shared/src/components/profile/HeroImage.tsx [hunk 23]
```diff
@@ -0,0 +1,38 @@
...
+  ref: Ref<HTMLDivElement>,
+): ReactElement {
+  return (
+    <div className="flex relative mx-4 h-24" ref={ref}>
+      {cover ? (
+        <Image
+          src={cover}
+          alt="cover"
+          loading="eager"
+          className="object-cover absolute top-0 left-0 -z-1 w-full h-full rounded-26"
```
Comment (rebelchris) [inline packages/shared/src/components/profile/HeroImage.tsx:21]: We should now have access to `size-full` in favor of `w-full + h-full`

File: packages/shared/src/components/profile/HeroImage.tsx [hunk 24]
```diff
@@ -0,0 +1,38 @@
...
+        />
+      ) : (
+        <div className="absolute top-0 left-0 -z-1 w-full h-full rounded-26 bg-theme-bg-secondary" />
```
Comment (rebelchris) [inline packages/shared/src/components/profile/HeroImage.tsx:24]: Same here:
We should now have access to `size-full` in favor of `w-full + h-full`

File: packages/shared/src/components/profile/ProfileWidgets.tsx [hunk 25]
```diff
@@ -0,0 +1,101 @@
...
+      className={classNames('flex flex-col gap-6 my-4 tablet:my-0', className)}
+    >
+      <Header user={user} isSameUser={isSameUser} className="-mb-2" />
+      {!hideSticky && (
+        <Header
+          user={user}
+          isSameUser={isSameUser}
+          sticky
+          className="fixed top-0 left-0 z-3 w-full transition-transform duration-75 bg-theme-bg-primary"
+          style={{ transform: `translateY(${(stickyProgress - 1) * 100}%)` }}
```
Comment (rebelchris) [inline packages/shared/src/components/profile/ProfileWidgets.tsx:54]: Ah I see :).

File: packages/shared/src/components/profile/RanksWidget.tsx [hunk 26]
```diff
@@ -0,0 +1,73 @@
...
+const RankHistory = ({
+  rank,
+  rankName,
+  count,
+}: RankHistoryProps): ReactElement => (
+  <div
+    className="flex flex-col tablet:flex-row items-center p-2 tablet:py-1 tablet:pr-4 tablet:pl-2 font-bold rounded-12 border typo-callout border-theme-bg-secondary"
+    aria-label={`${rankName}: ${count}`}
+  >
+    <Rank className="w-8 h-8" rank={rank} colorByRank />
```
Comment (rebelchris) [inline packages/shared/src/components/profile/RanksWidget.tsx:26]: Should also have `size-8` now

File: packages/shared/src/components/profile/Readme.tsx (hunk 9 above)
Comment (rebelchris) [reply packages/shared/src/components/profile/Readme.tsx:39]: Optional could have had these in a useProfile hook or something, but not blocking the PR

File: packages/shared/src/components/profile/Readme.tsx [hunk 27]
```diff
@@ -0,0 +1,130 @@
...
+      }),
+    {
+      enabled: editMode,
+      refetchOnWindowFocus: false,
+      refetchOnReconnect: false,
+      refetchOnMount: false,
+    },
+  );
+
+  const { mutateAsync: updateReadme, isLoading: submitting } = useMutation<
```
Comment (rebelchris) [inline packages/shared/src/components/profile/Readme.tsx:55]: This part as well in that case above of a hook

File: packages/shared/src/components/profile/Readme.tsx [hunk 28]
```diff
@@ -0,0 +1,130 @@
...
+      if (!message) {
+        return;
+      }
+
+      displayToast(message);
+    },
+  });
+
+  let children: ReactNode;
+  if (editMode) {
```
Comment (rebelchris) [inline packages/shared/src/components/profile/Readme.tsx:79]: This feels a bit messy to me, should we not prefer 2 components in this case?
1 view and 1 edit one.

File: packages/shared/src/components/profile/SocialChips.tsx [hunk 29]
```diff
@@ -0,0 +1,72 @@
...
+    label: (x) => `@${x}`,
+  },
+  portfolio: {
+    icon: <LinkIcon />,
+    href: (x) => x,
+    // Strip protocol from url
+    label: (x) => x.replace(/(^\w+:|^)\/\//, ''),
+  },
+};
+const order: (keyof SocialChipsProps['links'])[] = [
```
Comment (rebelchris) [inline packages/shared/src/components/profile/SocialChips.tsx:36]: could also have added `order` in the above, but this is fine to me.

File: packages/shared/src/components/profile/SocialChips.tsx [hunk 30]
```diff
@@ -0,0 +1,72 @@
...
+  'github',
+  'twitter',
+  'portfolio',
+];
+
+export function SocialChips({ links }: SocialChipsProps): ReactElement {
+  const elements = order
+    .filter((key) => !!links[key])
+    .map((key) => (
+      <Button
```
Comment (rebelchris) [inline packages/shared/src/components/profile/SocialChips.tsx:46]: Did you choose the old button v1 on purpose?
Technically it should be unused soon as we plan to remove it.

File: packages/shared/src/components/profile/SquadsList.tsx [hunk 31]
```diff
@@ -0,0 +1,204 @@
...
+
+      return;
+    }
+
+    await joinSquad();
+  };
+
+  return (
+    <div
+      className="flex relative items-center p-2 first:ml-4 tablet:first:ml-0 w-40 tablet:w-auto bg-theme-float hover:bg-theme-hover rounded-2xl"
```
Comment (rebelchris) [inline packages/shared/src/components/profile/SquadsList.tsx:78]: `rounded-2xl` should be the numeric version not sure how much it comes to but something like `rounded-12`

File: packages/shared/src/components/profile/SquadsList.tsx [hunk 32]
```diff
@@ -0,0 +1,204 @@
...
+    >
+      <Link href={squad.permalink} prefetch={false} passHref>
+        <CardLink />
+      </Link>
+      <div className="flex overflow-hidden flex-col flex-1">
+        <div className="flex gap-2 items-center">
+          <Image
+            src={squad.image}
+            alt={squad.name}
+            className="w-8 h-8 rounded-full"
```
Comment (rebelchris) [inline packages/shared/src/components/profile/SquadsList.tsx:89]: size-8 should do

File: packages/shared/src/components/profile/SquadsList.tsx [hunk 33]
```diff
@@ -0,0 +1,204 @@
...
+          />
+          <div className="flex flex-col flex-1">
+            <div className="overflow-hidden font-bold whitespace-nowrap typo-caption1 text-ellipsis">
+              {squad.name}
+            </div>
+            <div className="overflow-hidden whitespace-nowrap text-theme-label-quaternary typo-caption2 text-ellipsis">
+              @{squad.handle}
+            </div>
```
Comment (rebelchris) [inline packages/shared/src/components/profile/SquadsList.tsx:97]: These should be divs or can we get away with paragraphs?

File: packages/shared/src/components/profile/SquadsList.tsx [hunk 34]
```diff
@@ -0,0 +1,204 @@
...
+          </span>
+          {showJoin && (
+            <Button
+              className="tablet:hidden z-1 ml-auto"
+              variant={ButtonVariant.Secondary}
+              size={ButtonSize.XSmall}
+              icon={<PlusIcon />}
+              onClick={onJoin}
+              loading={isLoading}
+              aria-label="Join Squad"
```
Comment (rebelchris) [inline packages/shared/src/components/profile/SquadsList.tsx:122]: Don't think you need this specifically? (but could mistake it)

File: packages/shared/src/components/profile/SquadsList.tsx [hunk 35]
```diff
@@ -0,0 +1,204 @@
...
+        <Button
+          variant={ButtonVariant.Float}
+          size={ButtonSize.Large}
+          tag="a"
+          href={`${webappUrl}squads/new?origin=profile`}
+          icon={<PlusIcon />}
+          aria-label="Create a new Squad"
+        />
+        <div className="rounded-2xl border flex-[3] border-theme-divider-tertiary" />
+        <div className="rounded-l-2xl border flex-[2] border-theme-divider-tertiary" />
```
Comment (rebelchris) [inline packages/shared/src/components/profile/SquadsList.tsx:183]: These can also have the radius numeric preferably

File: packages/shared/src/hooks/profile/useActivityTimeFilter.ts [hunk 36]
```diff
@@ -0,0 +1,58 @@
...
+      const start = startOfTomorrow();
+      return [start, subYears(subDays(start, 2), 1)];
+    }
+    const selected = parseInt(dropdownOptions[selectedHistoryYear], 10);
+    const startYear = new Date(selected, 0, 1);
+
+    return [addDays(endOfYear(startYear), 1), startYear];
+    // @NOTE see https://dailydotdev.atlassian.net/l/cp/dK9h1zoM
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [selectedHistoryYear]);
```
Comment (rebelchris) [inline packages/shared/src/hooks/profile/useActivityTimeFilter.ts:48]: Should we not add all deps here to avoid breaking the rules.
Shouldn't make too much difference I think.

File: packages/webapp/components/layouts/ProfileLayout/v2.tsx [hunk 37]
```diff
@@ -0,0 +1,142 @@
+import React, { ReactElement, ReactNode } from 'react';
```
Comment (rebelchris) [inline packages/webapp/components/layouts/ProfileLayout/v2.tsx:1]: With removal of index should we not just call this `index`?

File: packages/webapp/pages/[userId]/index.tsx [hunk 38]
```diff
@@ -1,306 +1,104 @@
...
+    generateQueryKey(RequestKey.Profile, initialUser),
+    () => getProfile(initialUser?.id),
+    {
+      placeholderData: initialUser,
+      refetchOnReconnect: false,
+      refetchOnMount: false,
+      refetchOnWindowFocus: false,
+      // Make sure the logged in user's profile is always up to date
+      enabled: !!initialUser && isSameUser,
```
Comment (rebelchris) [inline packages/webapp/pages/[userId]/index.tsx:52]: This would just mean it will always refetch even if you didn't change, not needed for this PR, but actually had a thought about it.

1. Can we not manually manipulate the cache as we do in other places, that should work I think
2. If issues we could actually start using back redirects with query params like ?r=true as when `r` is set we refresh enable?

Although think we prefer 1 with a staleTime?

**Review by rebelchris — CHANGES_REQUESTED** [review]
Amazing amount of work!
Some minor comments, you can take some with a grain of salt as suggestions :)

File: packages/shared/src/components/ProfilePicture.tsx (hunk 16 above)
Comment (idoshamun) [reply packages/shared/src/components/ProfilePicture.tsx:107]: Wdym? It's still native lazy loading, but I wanted to add an option to eagerly load them. The default is lazy load but to optimize the profile page, I wanted above the fold images to load immediately

File: packages/shared/src/components/profile/Header.tsx (hunk 20 above)
Comment (idoshamun) [reply packages/shared/src/components/profile/Header.tsx:44]: It's not a `p` really but maybe a `span` would be better then. Right?

File: packages/shared/src/components/profile/Header.tsx (hunk 21 above)
Comment (idoshamun) [reply packages/shared/src/components/profile/Header.tsx:66]: TypeScript complains because onClick has some args

File: packages/shared/src/components/profile/Header.tsx (hunk 22 above)
Comment (idoshamun) [reply packages/shared/src/components/profile/Header.tsx:76]: Wdym?

File: packages/shared/src/components/profile/HeroImage.tsx (hunk 23 above)
Comment (idoshamun) [reply packages/shared/src/components/profile/HeroImage.tsx:21]: Neat! Didn't know that

File: packages/shared/src/components/profile/Header.tsx (hunk 19 above)
Comment (idoshamun) [reply packages/shared/src/components/profile/Header.tsx:35]: I see that you already figured it out. So I guess it's ok :)

File: packages/shared/src/components/profile/SocialChips.tsx (hunk 30 above)
Comment (idoshamun) [reply packages/shared/src/components/profile/SocialChips.tsx:46]: Yes I copied an existing implementation that Tsahi referred me to. We'll be able to migrate this button only when we introduce the "Subtle" variation. Don't know the timeline for that though

File: packages/shared/src/components/ProfilePicture.tsx (hunk 16 above)
Comment (rebelchris) [reply packages/shared/src/components/ProfilePicture.tsx:107]: We try to avoid the `lazy` one as it uses js lazy loading (from top of my head) we have a option for `nativeLazyLoading` which uses plain `img` elements props.

File: packages/shared/src/components/profile/SquadsList.tsx (hunk 33 above)
Comment (idoshamun) [reply packages/shared/src/components/profile/SquadsList.tsx:97]: Span perhaps?

File: packages/shared/src/components/profile/SquadsList.tsx (hunk 34 above)
Comment (idoshamun) [reply packages/shared/src/components/profile/SquadsList.tsx:122]: It's needed for accessibility since this button has nothing but icon

File: packages/shared/src/components/profile/Header.tsx (hunk 20 above)
Comment (rebelchris) [reply packages/shared/src/components/profile/Header.tsx:44]: To me text is most of the time `p` but not sure how others view it.
Span could also work, but I tend to use it for inline things because of it's native inline behaviour

File: packages/shared/src/components/profile/Header.tsx (hunk 21 above)
Comment (rebelchris) [reply packages/shared/src/components/profile/Header.tsx:66]: Ah yes seen this before :) nvm then

File: packages/shared/src/hooks/profile/useActivityTimeFilter.ts (hunk 36 above)
Comment (idoshamun) [reply packages/shared/src/hooks/profile/useActivityTimeFilter.ts:48]: I can. Just copied whatever existed into a standalone hook. Didn't touch it

File: packages/webapp/components/layouts/ProfileLayout/v2.tsx (hunk 37 above)
Comment (idoshamun) [reply packages/webapp/components/layouts/ProfileLayout/v2.tsx:1]: Good catch!

File: packages/shared/src/components/profile/SocialChips.tsx (hunk 30 above)
Comment (rebelchris) [reply packages/shared/src/components/profile/SocialChips.tsx:46]: Ah ok got it :) 
Yeah there are 3/4 exclusions now we need to find out still so no problem for this PR.

File: packages/webapp/pages/[userId]/index.tsx (hunk 38 above)
Comment (idoshamun) [reply packages/webapp/pages/[userId]/index.tsx:52]: This will not fix all the caching issues I'm trying to solve. The primary one is given we use next ISR, the page is cached for quite some time. But let's say I just updated my readme and then refresh the page, without this I'll see the old readme. With this I see always fresh data

Comment (rebelchris) [reply packages/webapp/pages/[userId]/index.tsx:52]: Ah yes, but in that case would the 2nd option not work of leveraging query params to force refreshes?
(it's not really blocking, more opening the discussion for future things as we have quite a few of these issues already)

Comment (idoshamun) [reply packages/webapp/pages/[userId]/index.tsx:52]: When the user manually decides to visit their profile again I cannot force them into adding query param, right? Or am I missing something?

File: packages/shared/src/components/ProfilePicture.tsx (hunk 16 above)
Comment (idoshamun) [reply packages/shared/src/components/ProfilePicture.tsx:107]: I believe this is exactly what we do here

File: packages/webapp/pages/[userId]/index.tsx (hunk 38 above)
Comment (rebelchris) [reply packages/webapp/pages/[userId]/index.tsx:52]: That case no, my suggestion was:
- user edit profile
- on save we invoke toast + redirect back to profile + ?r=true

(but now I'm not sure if we didn't redirect for specific reason maybe)

File: packages/shared/src/components/ProfilePicture.tsx (hunk 16 above)
Comment (rebelchris) [reply packages/shared/src/components/ProfilePicture.tsx:107]: Oh sigh, I for some reason thought this was the bottom evaluation 🤦

File: packages/webapp/pages/[userId]/index.tsx (hunk 38 above)
Comment (idoshamun) [reply packages/webapp/pages/[userId]/index.tsx:52]: I feel like this is not a coherent solution for my use case then. If that's ok can we keep it as is for now?

Comment (rebelchris) [reply packages/webapp/pages/[userId]/index.tsx:52]: Yeah 100% ok for this PR :)

File: packages/shared/src/hooks/profile/useProfile.ts [hunk 39]
```diff
@@ -0,0 +1,25 @@
...
+
+export function useProfile(initialUser?: PublicProfile): PublicProfile {
+  const { user: loggedUser } = useContext(AuthContext);
+  const isSameUser = loggedUser?.id === initialUser?.id;
+
+  const { data: user } = useQuery(
+    generateQueryKey(RequestKey.Profile, initialUser),
+    () => getProfile(initialUser?.id),
+    {
+      placeholderData: initialUser,
```
Comment (rebelchris) [inline packages/shared/src/hooks/profile/useProfile.ts:15]: This initialUser is already fetched data right, I think you might want to use `initialData` in that case?
It's a small nuance, but initial will persist already.

https://tkdodo.eu/blog/placeholder-and-initial-data-in-react-query

File: packages/shared/src/components/profile/Header.tsx (hunk 22 above)
Comment (rebelchris) [reply packages/shared/src/components/profile/Header.tsx:76]: Ah no never mind, I thought we added aria-label automatically in the button, but we don't 😢

**Review by rebelchris — APPROVED** [review]
Nothing blocking from my side 🚀

File: packages/shared/src/hooks/profile/useProfile.ts (hunk 39 above)
Comment (idoshamun) [reply packages/shared/src/hooks/profile/useProfile.ts:15]: My default is initial data but here I specifically wanted placeholder to make sure I fetch fresh data for a specific use case. Otherwise, I may have shown cached data to the user
